src/ray/common/task/task_spec.cc

```diff
@@ -268,11 +268,19 @@ void TaskSpecification::AddDynamicReturnId(const ObjectID &dynamic_return_id) {
 }
 
 bool TaskSpecification::ArgByRef(size_t arg_index) const {
+  return message_->args(arg_index).has_object_ref() &&
+         !message_->args(arg_index).is_inlined();
+}
+
+bool TaskSpecification::HasArgRef(size_t arg_index) const {
   return message_->args(arg_index).has_object_ref();
```

Comment:
Should we check `is_inlined` in this method?

Reply:
We have already checked `is_inlined` in `ArgByRef`. I have already removed `HasArgRef`.